File: 643_MaximumAverageSubarrayI/maximumAverageSubarrayI.js
var findMaxAverage = function(nums, k) {
	let sum = 0
	for (let i = 0; i < k; i++) {
		sum += nums[i]
	}

	let ans = sum
	for (let i = k; i < n; i++) {
		sum += num[i] - nums[i - k]
		ans = Math.max(ans, sum)
	}

	return ans / k
}

console.log(findMaxAverage([1,12,-5,-6,50,3], 4))
console.log(findMaxAverage([5], 1))

File: 643_MaximumAverageSubarrayI/maximumAverageSubarrayI.py
def findMaxAverage(self, nums, k):
    sum = 0
    for i in range(k):
        sum += nums[i]

    ans = sum
    for i in range(k, len(nums)):
        sum += nums[i] - nums[i - k]
        ans = max(ans, sum)

    return ans / k

print(findMaxAverage([1,12,-5,-6,50,3], 4))
print(findMaxAverage([5], 1))